app/scripts/modules/amazon/src/domain/IAmazonFunction.ts

```diff
@@ -0,0 +1,29 @@
+import { IFunction } from '@spinnaker/core';
+
+export interface IAmazonFunction extends IFunction {
+  credentials?: string;
+  role?: string;
+  runtime: string;
+  s3bucket: string;
+  s3key: string;
+  handler: string;
+  functionName: string;
+  publish: boolean;
+  description: string;
+  tags: [{}];
+  memorySize: number;
+  timeout: number;
+  envVariables: {};
```

Comment:
If `envVariables` will not be empty it would be stronger to make the type `{ [key: string]: any }`.